### DISCUSSION TITLE
Using JFNK and PBP to solve an 3D CFD problem

### DISCUSSION THREAD

**Discussion by js-jixu** [idaholab/moose#23496] [Q&A Modules: Navier-Stokes] [answered]
Hi, experts.

I have seen many examples of applying JFNK method to CFD, so I want to use MOOSE to do some 3D CFD calculations. But when I use `JFNK` algorithm and `PBP` in MOOSE to calculate a 3D-CFD case, the result cannot converge. When I use `NEWTON` method with `FSP` or `SMP`, it can converge well. Here are my Preconditioning and Executioner settings:

```
[Preconditioning]
  active = 'PBP'
  [PBP]
    type = PBP
    solve_order =    'p   velocity Tf       Ts'
    preconditioner = 'LU  LU       LU       LU'
    full = true
    petsc_options_iname = '-ksp_type'
    petsc_options_value = 'gmres'
  []
[]

[Executioner]
  type = Steady
  solve_type = 'JFNK'
  petsc_options = '-snes_converged_reason -ksp_converged_reason -snes_linesearch_monitor'
  line_search = 'none'

  nl_abs_tol = 1e-8
  nl_rel_tol = 1e-8
  nl_max_its = 50
  l_max_its = 100
[]
```

Do you have any suggestions or similar experiences?

**Comment by GiudGiud** [comment, marked as answer]
Use PJFNK, JFNK is meant to avoid preconditioning

**Reply by lindsayad** [reply]
PBP was meant to be used with JFNK, not PJFNK. But I would say that you should use FSP instead of PBP
